sda-commons-shared-otel-instrumentation/src/main/java/org/sdase/commons/shared/instrumentation/AgentLoader.java

```diff
@@ -0,0 +1,36 @@
+package org.sdase.commons.shared.instrumentation;
+
+import io.opentelemetry.contrib.attach.RuntimeAttach;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This utils is only responsible for loading the openTelemetry agent. All configuration is already
+ * done through environment variables.
+ */
+public class AgentLoader {
+  private AgentLoader() {
+    // utility class
+  }
+
+  private static final Logger LOG = LoggerFactory.getLogger(AgentLoader.class);
+
+  public static void load() {
+    String jaegerSamplerType = System.getenv("JAEGER_SAMPLER_TYPE");
+    String jaegerSamplerParam = System.getenv("JAEGER_SAMPLER_PARAM");
```

Comment:
You should use `new SystemPropertyAndEnvironmentLookup().lookup(...)` instead to make testing / configuration a little easier. It's not possible to replace env variables in tests starting from Java 16.

Reply:
I changed it for this case, but for the agent module, maybe we can skip adding the dependency, and anyways internally it is using `System.getenv`, also replace env variables in tests is not an option, because the agent does most of the configuration at startup, and updates won't be picked up.